src/OrbitGl/TrackManagerTest.cpp

```diff
@@ -0,0 +1,135 @@
+      nullptr, orbit_grpc_protos::CaptureStarted{}, std::nullopt, absl::flat_hash_set<uint64_t>{});
+
+  // AddressInfo
+  orbit_client_protos::LinuxAddressInfo address_info;
+  address_info.set_absolute_address(kInstructionAbsoluteAddress);
+  address_info.set_offset_in_function(kInstructionAbsoluteAddress - kFunctionAbsoluteAddress);
+  address_info.set_function_name(kFunctionName);
+  address_info.set_module_path(kModuleName);
+  capture_data->InsertAddressInfo(address_info);
+
+  // CallstackInfo
+  const std::vector<uint64_t> callstack_frames{kInstructionAbsoluteAddress};
+  orbit_client_protos::CallstackInfo callstack_info;
+  *callstack_info.mutable_frames() = {callstack_frames.begin(), callstack_frames.end()};
+  callstack_info.set_type(orbit_client_protos::CallstackInfo_CallstackType_kComplete);
+  capture_data->AddUniqueCallstack(kCallstackId, std::move(callstack_info));
+
+  // CallstackEvent
+  orbit_client_protos::CallstackEvent callstack_event;
+  callstack_event.set_callstack_id(kCallstackId);
+  callstack_event.set_thread_id(kThreadId);
+  capture_data->AddCallstackEvent(std::move(callstack_event));
+
+  capture_data->AddOrAssignThreadName(kThreadId, kThreadName);
+
+  return capture_data;
+}
+
+}  // namespace
+
+namespace orbit_gl {
+
+class UnitTestTrack : public Track {
+ public:
+  explicit UnitTestTrack(TimeGraphLayout* layout, Track::Type type, const std::string& name)
+      : Track(nullptr, nullptr, nullptr, layout, nullptr, 0), name_(name), type_(type){};
+
+  [[nodiscard]] Type GetType() const override { return type_; }
+  [[nodiscard]] float GetHeight() const override { return 100.f; }
+  [[nodiscard]] bool IsEmpty() const override { return false; }
+
+  [[nodiscard]] std::vector<std::shared_ptr<TimerChain>> GetAllChains() const override {
+    return {};
+  }
+  [[nodiscard]] std::vector<std::shared_ptr<TimerChain>> GetAllSerializableChains() const override {
+    return {};
+  }
+
+ private:
+  std::string name_;
+  Type type_;
+};  // namespace orbit_gl
+
+class TrackManagerTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    capture_data_ = GenerateTestCaptureData();
+    track_manager_ =
+        std::make_unique<TrackManager>(nullptr, nullptr, &layout_, nullptr, capture_data_.get());
+  }
+
+  void CreateAndFillTracks() {
+    auto* scheduler_track = track_manager_->GetOrCreateSchedulerTrack();
+    auto* thread_track = track_manager_->GetOrCreateThreadTrack(kThreadId);
+
+    TimerInfo timer;
+    timer.set_start(0);
+    timer.set_end(100);
+    timer.set_thread_id(kThreadId);
+    timer.set_processor(0);
+    timer.set_depth(0);
+    timer.set_type(TimerInfo::kCoreActivity);
+
+    scheduler_track->OnTimer(timer);
+    timer.set_type(TimerInfo::kCoreActivity);
+    thread_track->OnTimer(timer);
+  }
+
+  TimeGraphLayout layout_;
+  std::unique_ptr<orbit_client_model::CaptureData> capture_data_;
+  std::unique_ptr<TrackManager> track_manager_;
```

Comment:
You could consider using an optional since the only "feature" from unique_ptr you're using is the optionality.

Reply:
Actually this was due to the initialization happening in `SetUp`, but I've changed this to the constructor anyways, so this changed to `TrackManager track_manager_`, it does not need to be optional.